components/clarinet-cli/src/frontend/cli.rs

```diff
@@ -395,9 +398,13 @@ struct Integrate {
 
 #[derive(Parser, PartialEq, Clone, Debug)]
 struct Test {
-    /// Generate coverage file (coverage.lcov)
-    #[clap(long = "coverage")]
-    pub coverage: bool,
+    /// Generate coverage file, and optionally provide name of generated file (defaults to "coverage.lcov")
+    #[clap(
+        long = "coverage",
+        default_missing_value("coverage.lcov"),
```

Comment:
The `clap::default_missing_value` function allows the legacy functionality to still work. A user can use just the `--coverage` flag, and it will default to `"coverage.lcov"`.